File: ank-gradle-plugin/src/main/java/com/kategory/gradle/plugin/ank/AnkPlugin.java
package com.kategory.gradle.plugin.ank;

import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.gradle.api.tasks.SourceSetContainer;

public class AnkPlugin implements Plugin<Project> {
  static final String TASK_NAME = "ank";

  @Override
  public void apply(Project target) {
    // TODO This is a test!
    SourceSetContainer sourceSets = (SourceSetContainer)
      target.getProperties().get("sourceSets");
    sourceSets.getByName("main").getRuntimeClasspath();

    target.getTasks().create(TASK_NAME, AnkTask.class);
  }
}

File: ank-plugin/plugin/src/main/kotlin/plugin/AnkPlugin.kt
package plugin

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.Task
import org.gradle.api.tasks.Exec

import org.gradle.kotlin.dsl.*

open class AnkPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        project.run {

            tasks {
                "myEchoTask"(Exec::class) {
                    executable("sh")
                    args("-c", "echo 'Hello Ank from your shell!'")
                }
            }
        }
    }
}
